Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/Mandate/Notification.hs

```diff
@@ -57,16 +57,31 @@ sendPDNNotificationToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId
         mbMerchantOpCityId = jobData.merchantOperatingCityId
         startTime = jobData.startTime
         endTime = jobData.endTime
+        retryCount = jobData.retryCount
     merchant <- CQM.findById merchantId >>= fromMaybeM (MerchantNotFound merchantId.getId)
     merchantOpCityId <- CQMOC.getMerchantOpCityId mbMerchantOpCityId merchant Nothing
     transporterConfig <- SCT.findByMerchantOpCityId merchantOpCityId >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
     let limit = transporterConfig.driverFeeMandateNotificationBatchSize
-    driverFees <- QDF.findDriverFeeInRangeWithNotifcationNotSentAndStatus merchantId limit startTime endTime DF.PAYMENT_PENDING
+    driverFees <- QDF.findDriverFeeInRangeWithNotifcationNotSentAndStatus merchantId limit startTime endTime retryCount DF.PAYMENT_PENDING
+    maxShards <- asks (.maxShards)
     if null driverFees
       then do
-        maxShards <- asks (.maxShards)
-        scheduleJobs transporterConfig startTime endTime merchantId merchantOpCityId maxShards
-        return Complete
+        driverFeesWithRetryContPlusOne <- QDF.findDriverFeeInRangeWithNotifcationNotSentAndStatus merchantId 1 startTime endTime (retryCount + 1) DF.PAYMENT_PENDING
+        if retryCount >= transporterConfig.notificationRetryCountThreshold && null driverFeesWithRetryContPlusOne
```

Comment:
Could you explain this?

Reply:
- On line 69 we are checking if there is a batch of failed notification with retryCount = currentRetryCount + 1
- If there is no such case OR we haven crossed retry threshold the then we schedule the executions else we do retry for retryCount = currentRetryCount + 1 (next retry batch)
PS:- I will put || instead of &&